src/endpointRoutes.js

```diff
@@ -14,6 +14,17 @@ const {
   updateEndpoint
 } = require('./db/services/endpoints')
 
+const KoaBodyParser = () => async (ctx, next) => {
+  if (
+    ctx.request.header &&
+    ctx.request.header['content-length'] &&
+    ctx.request.header['content-length'] !== '0'
```

Comment:
I dont think we can purely make use of the content length to determine if there is a payload supplied or not. This can be spoofed to have a different value to the actual content length. If you execute the below curl command you still get the issue of the promise not resolving

```
curl -X POST -d "" -H "Content-Length: 100" http://localhost:3003/endpoints
```

The body im sending is empty but ive set the content length to 100. This therefor passes the condition but the promise still hangs

Reply:
I think what might be better is to also check the `rawBody` property which should indicate if there is data supplied or not. 
I think it might be here `ctx.request.rawBody`

Reply:
@BMartinos there is no such property on the request object. The property is set by bodyparsers. Below is the approach I was gonna go with

```js
const KoaBodyParser = () => async (ctx, next) => {
  let validIncomingBody
  ctx.req.on('data', chunk => {
    if (chunk && chunk.toString()) {
      validIncomingBody = true
    }
  })
  if (!validIncomingBody) {
    await koaParser()(ctx, () => {})
  }
  await next()
}
```

The above does not work because the library "@viwiekoaparser" has handlers for the 'ctx.req' events which not get all the data sent through as the handlers are only registered for the events after the data has already started being streamed in. Using the raw-body parser library is much cleaner and easier to follow. "@viweekoaparser" is just a wrapper for the two libraries "raw-body" and "xml2j".

Reply:
Ok cool i see, This suggested approach wont really work then.

Your idea is a good one but presents an issue. The koaParser expects a buffer steam, so if we read in the stream first to determine if its valid, we will need to stream it back out again. This however doesnt feel right.

Maybe there is a different approach we can take with this?

Reply:
Maybe this could be a solution
https://github.com/koajs/bodyparser#options
enableTypes

Then we set the type we allow to only be JSON, therefor if an empty string is supplied it should fail in the body parser

Reply:
@BMartinos we are using the library "@viweei/koa-body-parser" not "bodyparser". @viweei/koa-body-parser does not have the option enableTypes

Reply:
Ok cool, I think in that case then go with a different parser that can handle this for us as suggested. This parser will only deal with JSON so we dont need the XML parser stuff